jetbrains-ultimate/src-201/software/aws/toolkits/jetbrains/services/rds/actions/CreateConfigurationAction.kt

```diff
@@ -0,0 +1,97 @@
+import software.aws.toolkits.jetbrains.services.rds.RdsNode
+import software.aws.toolkits.jetbrains.services.rds.auth.IamAuth
+import software.aws.toolkits.jetbrains.services.rds.auth.CREDENTIAL_ID_PROPERTY
+import software.aws.toolkits.jetbrains.services.rds.auth.REGION_ID_PROPERTY
+import software.aws.toolkits.jetbrains.services.rds.jdbcMysql
+import software.aws.toolkits.jetbrains.services.rds.jdbcPostgres
+import software.aws.toolkits.jetbrains.services.rds.mysqlEngineType
+import software.aws.toolkits.jetbrains.services.rds.postgresEngineType
+import software.aws.toolkits.jetbrains.services.rds.ui.CreateDataSourceDialogWrapper
+import software.aws.toolkits.jetbrains.utils.actions.OpenBrowserAction
+import software.aws.toolkits.jetbrains.utils.notifyError
+import software.aws.toolkits.resources.message
+
+// It is registered in ext-datagrip.xml FIX_WHEN_MIN_IS_201
+@Suppress("ComponentNotRegistered")
+class CreateDataSourceActionGroup : SingleExplorerNodeActionGroup<RdsNode>("TODO add config"), DumbAware {
+    override fun getChildren(selected: RdsNode, e: AnActionEvent): List<AnAction> = listOf(
+        CreateIamDataSourceAction(selected)
+    )
+}
+
+class CreateIamDataSourceAction(private val node: RdsNode) : AnAction(message("rds.iam_config")) {
+    override fun actionPerformed(e: AnActionEvent) {
+        if (!checkPrerequisites()) {
+            return
+        }
+        val dialog = CreateDataSourceDialogWrapper(node.nodeProject, node.dbInstance)
+        if (!dialog.showAndGet()) {
+            return
+        }
+        val username = dialog.getUsername() ?: throw IllegalStateException("Username is null, but it should have already been validated not null!")
+        val database = dialog.getDatabaseName()
+        createDatasource(username, database)
+    }
+
+    private fun checkPrerequisites(): Boolean {
+        // Assert IAM auth enabled
+        if (!node.dbInstance.iamDatabaseAuthenticationEnabled()) {
+            notifyError(
+                project = node.nodeProject,
+                title = message("aws.notification.title"),
+                content = message("validation_error.no_iam_auth", node.dbInstance.dbName()),
+                action = OpenBrowserAction(message("rds.validation.setup_guide"), null, HelpIds.RDS_SETUP_IAM_AUTH.url)
+            )
+            return false
+        }
+        return true
+    }
+
+    private fun createDatasource(username: String, database: String) {
+        val endpoint = node.dbInstance.endpoint()
+        val url = "${endpoint.address()}:${endpoint.port()}"
+        val source = LocalDataSourceManager.getInstance(node.nodeProject)
+        val dataSource = LocalDataSource().also {
+            // turn on url only config since we don't need any additional config
+            it.isConfiguredByUrl = true
+            it.authProviderId = IamAuth.providerId
+            it.additionalJdbcProperties[CREDENTIAL_ID_PROPERTY] = node.nodeProject.activeCredentialProvider().id
```

Comment:
Just wanted to note that, these properties won't be passed to jdbc (despite it's name), they are to be used by interceptor (I see that they are used, but anyway).